### PR TITLE
update @types/jenkins definition version 1.0.2

### PR DESCRIPTION
**Pull request by naoya7076**
The type definitions for [node-jenkin](https://github.com/silas/node-jenkins) have been updated to version 1.0 and above, and I have made corrections and updates due to the type inconsistencies that arose.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<[url here](https://github.com/silas/node-jenkins/tree/1.0.2)>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## jenkins ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jenkins@latest/))
was missing the following properties:
1. Jenkins
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 368ea13328d25f90d8a8e257adb686d21a702082
</p>

File: types/jenkins/index.d.ts [hunk 1]
```diff
@@ -1,162 +1,70 @@
...
+    name: string;
+    parameters?: any;
+    token?: string | undefined;
 }
 
-declare function create(opts?: {
-    baseUrl?: string | undefined;
-    crumbIssuer?: boolean | undefined;
-    headers?: any;
-    promisify?: false | undefined;
```
Comment (naoya7076) [inline types/jenkins/index.d.ts:154]: v1.0.2 does not support synchronous mode.

File: types/jenkins/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
     "compilerOptions": {
         "module": "commonjs",
         "lib": [
-            "es6"
+            "es6",
+            "dom"
```
Comment (naoya7076) [inline types/jenkins/tsconfig.json:6]: To use FormData type definition, add "dom".

Comment (jakebailey) [reply types/jenkins/tsconfig.json:6]: I don't think that adding `DOM` to a package named `node-jenkis` is a good thing. Do you mean to import `form-data` or something?

File: types/jenkins/index.d.ts [hunk 3]
```diff
@@ -1,162 +1,70 @@
-// Type definitions for jenkins 0.23
+// Type definitions for jenkins 1.0.2
 // Project: https://github.com/silas/node-jenkins
 // Definitions by: rhysd <https://rhysd.github.io>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
+interface JenkinsClientOptions {
+    baseUrl: string;
+    crumbIssuer?: boolean | undefined | `() => void`;
```
Comment (jakebailey) [inline types/jenkins/index.d.ts:9]: I don't think this string is what you intended to use here.

**Review by jakebailey — CHANGES_REQUESTED** [review]
See above; we definitely don't want to include DOM on a node package.

File: types/jenkins/tsconfig.json (hunk 2 above)
Comment (jakebailey) [reply types/jenkins/tsconfig.json:6]: I mean that there's a library with that name, but maybe my memory is bad. In any case, I'm pretty sure I have read code which is for node that has this type without DOM, e.g. node-fetch.

Comment (naoya7076) [reply types/jenkins/tsconfig.json:6]: I understand what you mean. I'll take a look at libraries like node-fetch and try to create form-data type definition without using DOM.

Comment (naoya7076) [reply types/jenkins/tsconfig.json]: updated FormData type definition to import `form-data` library.
[`371a7a1` (#66784)](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66784/commits/371a7a16d4429d15cb19afc37097165c404a9864)

File: types/jenkins/index.d.ts (hunk 3 above)
Comment (naoya7076) [reply types/jenkins/index.d.ts]: Fixed to clearly define that it is a function definition.
[`30d12bd` (#66784)](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66784/commits/30d12bdb91b136277ca1580500460808db32e2c1)

File: types/jenkins/tsconfig.json (hunk 2 above)
Comment (jakebailey) [reply types/jenkins/tsconfig.json]: Thanks, looks correct given upstream has this same version range in its package.json.

**Review by jakebailey — APPROVED** [review]

**Comment by naoya7076** [comment]
Ready to merge

**Pull request merged**